beacon_node/beacon_processor/src/lib.rs

```diff
@@ -366,394 +350,104 @@ impl DuplicateCache {
 }
 
 /// An event to be processed by the manager task.
-#[derive(Derivative)]
-#[derivative(Debug(bound = "T: BeaconChainTypes"))]
-pub struct WorkEvent<T: BeaconChainTypes> {
-    drop_during_sync: bool,
-    work: Work<T>,
+#[derive(Debug)]
+pub struct WorkEvent<E: EthSpec> {
+    pub drop_during_sync: bool,
+    pub work: Work<E>,
 }
 
-impl<T: BeaconChainTypes> WorkEvent<T> {
```

Comment:
These `WorkEvent` constructors are no longer necessary. We're now just constructing the `WorkEvent` objects in the `beacon_node/network` crate directly.